test/extended/marketplace/source_create.go

```diff
@@ -0,0 +1,128 @@
+package marketplace
+
+import (
+	"time"
+
+	g "github.com/onsi/ginkgo"
+	o "github.com/onsi/gomega"
+
+	exutil "github.com/openshift/origin/test/extended/util"
+)
+
+var _ = g.Describe("[Feature:Marketplace] Marketplace", func() {
+	defer g.GinkgoRecover()
+
+	var (
+		oc            = exutil.NewCLI("marketplace", exutil.KubeConfigPath())
+		allNs         = "openshift-operators"
+		marketplaceNs = "openshift-marketplace"
+	)
+
+	g.AfterEach(func() {
+		//clear the sub,csv resource
+		allresourcelist := map[int][]string{
+			0: {"operatorsource", "opsrctest", marketplaceNs},
+			1: {"catalogsourceconfig", "csctest", marketplaceNs},
+			2: {"csv", "descheduler.v0.0.3", allNs},
+			3: {"subscription", "descheduler-test", allNs},
+		}
+		for _, source := range allresourcelist {
+			err := clearResources(oc, source[0], source[1], source[2])
+			o.Expect(err).NotTo(o.HaveOccurred())
+		}
+	})
+
+	//OCP-21405 create one opsrc,OCP-21419 create one csc,OCP-21479 sub to openshift-operators,OCP-21667 delete the csc
+	g.It("[ocp-21405 21419 21479 21667]create and delete the basic source,sub one operator", func() {
+		opsrcYaml := exutil.FixturePath("testdata", "marketplace", "opsrc", "01-opsrc.yaml")
+		err := createResources(oc, opsrcYaml)
+		o.Expect(err).NotTo(o.HaveOccurred())
+		time.Sleep(30 * time.Second)
+
+		opsrcResourceList := map[int][]string{
```

Comment:
Because of the `map` is unordered, so it's better to replace it with the `slice`.